### PR TITLE
Allow load all with page internally.

### PR DESCRIPTION
**Pull request by utas-raymondng**

### CODE REVIEW

File: server/src/main/java/au/org/aodn/ogcapi/server/core/service/ElasticSearch.java [hunk 1]
```diff
@@ -215,74 +221,153 @@ protected List<Hit<CategorySuggestDTO>> getCategorySuggestions(String input) thr
...
+                        catch(IOException ieo) {
+                            throw new RuntimeException(ieo);
+                        }
+                    }
+                }
+
+                @Override
+                public T next() {
+                    count.incrementAndGet();
+                    return response.get().hits().hits().get(index++).source();
```
Comment (vietnguyengit) [inline server/src/main/java/au/org/aodn/ogcapi/server/core/service/ElasticSearch.java:357]: how is this related to the PR yesterday that you thought it was a typo? https://github.com/aodn/es-indexer/pull/87

File: server/src/main/java/au/org/aodn/ogcapi/server/core/service/ElasticSearch.java [hunk 2]
```diff
@@ -215,74 +221,153 @@ protected List<Hit<CategorySuggestDTO>> getCategorySuggestions(String input) thr
...
+                    .query(q -> q.bool(createBoolQueryForProperties(queries, should, filters)))
+                    .sort(so -> so.score(v -> v.order(SortOrder.Desc)))
+                    .sort(so -> so
+                            .field(FieldSort.of(f -> f
+                                    .field(StacSummeries.Score.searchField)
+                                    .order(SortOrder.Desc))))
+                    .sort(so -> so
+                            // We need a unique key for the search, so _id is the best
+                            .field(FieldSort.of(f -> f
+                                    .field("id.keyword")
```
Comment (vietnguyengit) [inline server/src/main/java/au/org/aodn/ogcapi/server/core/service/ElasticSearch.java:254]: I'm not clear here, the id values are UUID random code rather than incremental digits or ordered alphabet characters, how can it help by sorting them either ASC or DESC?

Comment (utas-raymondng) [reply server/src/main/java/au/org/aodn/ogcapi/server/core/service/ElasticSearch.java:254]: The sort is based on the two above criteria first, then if they have both order, then last order is id, the way elastic recommend is you keep a fix sorting order for a particular query that can have record later then 2K, then the 2001 record is obtained by using the same sorting order query plus the search_after, in this case the id. 

The two initial criteria is not unique enough as they can have the same mathematical value, hence you need something unique.

File: server/src/main/java/au/org/aodn/ogcapi/server/core/service/ElasticSearch.java (hunk 1 above)
Comment (utas-raymondng) [reply server/src/main/java/au/org/aodn/ogcapi/server/core/service/ElasticSearch.java:357]: No this one is fine because the index++ is apply after the .get(), yesterday PR is fix because the index is pass in as a value, hence there is no point to do ++, after the call.

**Review by vietnguyengit — APPROVED** [review]
questions are explained in person, LGTM
